config/environments/test.rb

```diff
@@ -4,7 +4,7 @@
   config.action_view.cache_template_loading = true
   config.eager_load = false
   config.public_file_server.enabled = true
-  config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
+  config.public_file_server.headers = { 'Cache-Control' => 'public, no-cache, must-revalidate' }
```

Comment:
I'm cautious of the impact of this change on CI test performance. I think it's good for the ChromeDriver to cache assets if we don't expect them to change (such as in the context of a clean CI build).

Alternatively, could we find a way to limit these headers to apply only in local development? Or could we consider finding a way to cache-bust assets more aggressively in local development through methods like [filename-based fingerprint hashes](https://github.com/18F/identity-idp/blob/0abef7c8ddb5eb32b9e444e8a3badbf7a10a90a9/webpack.config.js#L14)?

Reply:
I think you are right to be concerned about pipeline performance here. You've made good suggestions, to which I'll add:

1. Maybe we can find a way to clear the Capybara cache (rather than just ask for no caching with headers)
2. Are non-static things being cached that shouldn't be?

Reply:
I could imagine that there might be some caching issues in local test runs if Capybara/Chromedriver doesn't guarantee a fresh session each time, since as observed in the link I shared previously, we only add the filename "fingerprint" in production environments, and this works in normal development because we [do disable caching for the development server](https://github.com/18F/identity-idp/blob/189e84ca1809b6ea17cc6e43f44de7ef73d83ae8/config/environments/development.rb#L48).

If that were the case, then yeah, I think I'd be interested in either finding way to delete that cache at the beginning of the test run, or conditionally setting these headers for local development testing vs. CI (e.g. maybe based on `ENV['CI']`), or always adding the filename fingerprint regardless of environment.